contracts/ceros/provider/BaseLpTokenProvider.sol

```diff
@@ -0,0 +1,384 @@
+        external
+        virtual
+        nonReentrant
+        whenNotPaused
+        onlyRole(PROXY)
+    {
+        require(_account != address(0));
+        _syncLp(_account);
+        _burnLp(_account, _amount);
+    }
+
+    /**
+     * @dev to make sure existing users who do not have enough lpToken can still burn
+     * only the available amount excluding delegated part will be burned
+     *
+     * @param _account lp token holder
+     * @param _amount amount to burn
+     */
+    function _safeBurnLp(address _account, uint256 _amount) virtual internal {
+        uint256 availableBalance = userLp[_account] - delegation[_account].amount;
+        if (_amount <= availableBalance) {
+            lpToken.burn(_account, _amount);
+        } else if (availableBalance > 0) {
+            // existing users do not have enough lpToken
+            lpToken.burn(_account, availableBalance);
+        }
+    }
+
+    /**
+     * @dev considering burn and mint with delta requires more if-else branches
+     * so here just burn all and re-mint all
+     *
+     * @param _account user address to sync
+     */
+    function _syncLp(address _account) internal virtual returns (bool) {
+        uint256 userExpectLp = dao.locked(token, _account);
+        uint256 userCurrentLp = userLp[_account];
+        if (userExpectLp == userCurrentLp) {
+            return false;
+        }
+
+        Delegation storage currentDelegation = delegation[_account];
+        uint256 currentDelegateLp = currentDelegation.amount;
+        address currentDelegateTo = currentDelegation.delegateTo;
+        // Step 1. burn all tokens
+        if (currentDelegateLp > 0) {
+            // burn delegatee's token
+            lpToken.burn(currentDelegateTo, currentDelegateLp);
+            currentDelegation.amount = 0;
+            // burn self's token
+            if (userCurrentLp > currentDelegateLp) {
+                _safeBurnLp(_account, userCurrentLp - currentDelegateLp);
+            }
+        } else {
+            _safeBurnLp(_account, userCurrentLp);
+        }
+
+        uint256 expectDelegateLp = userCurrentLp > 0 ? userExpectLp * currentDelegateLp / userCurrentLp : 0;
+        uint256 expectUserSelfLp = userExpectLp - expectDelegateLp;
+        if (expectDelegateLp > 0) {
+            lpToken.mint(currentDelegateTo, expectDelegateLp);
+            currentDelegation.amount = expectDelegateLp;
+        }
+        if (expectUserSelfLp > 0) {
+            lpToken.mint(_account, expectUserSelfLp);
+        }
+        userLp[_account] = userExpectLp;
+
+        emit SyncUserLp(_account, userExpectLp);
+        return true;
+    }
+
+    /**
+     * @dev change token address
+     * this method *should not* be called after user provides token through this provider
+     * as it will cause inconsistency between token and lpToken
+     * @param _token new token address
+     */
+    function changeToken(address _token) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
remove changeToken,changeLpToken,changeDao, the change will make the data inconsistency, we do not support this feature